apps/test/unit/templates/sectionProgressV2/SectionProgressSelectorTest.jsx

```diff
@@ -1,88 +1,175 @@
+    renderDefault();
+    store.dispatch(setShowProgressTableV2(true));
 
-    expect(wrapper.find('Link')).to.have.length(0);
+    expect(screen.queryByText(V1_PAGE_LINK_TEXT)).to.not.exist;
+    expect(screen.queryByText(V2_PAGE_LINK_TEXT)).to.not.exist;
   });
 
   it('shows v1 if disabled', () => {
     DCDO.set('progress-table-v2-enabled', false);
-    const wrapper = setUp({showProgressTableV2: true});
+    renderDefault();
+    store.dispatch(setShowProgressTableV2(true));
+
+    screen.getByTestId(V1_TEST_ID);
 
-    expect(wrapper.find(SectionProgress)).to.have.length(1);
-    expect(wrapper.find(SectionProgressV2)).to.have.length(0);
+    expect(screen.queryByText(V2_TEST_ID)).to.not.exist;
   });
 
   it('shows v1', () => {
-    const wrapper = setUp();
+    renderDefault();
 
-    expect(wrapper.find(SectionProgress)).to.have.length(1);
-    expect(wrapper.find(SectionProgressV2)).to.have.length(0);
-    expect(wrapper.find('Link')).to.have.length(1);
-    expect(wrapper.find('Link').at(0).props().children).to.equal(
-      'Switch to new progress view'
-    );
+    screen.getByText(V1_PAGE_LINK_TEXT);
+    screen.getByTestId(V1_TEST_ID);
+
+    expect(screen.queryByText(V2_PAGE_LINK_TEXT)).to.not.exist;
+    expect(screen.queryByText(V2_TEST_ID)).to.not.exist;
   });
 
   it('shows v2', () => {
-    const wrapper = setUp({showProgressTableV2: true});
+    renderDefault();
+    store.dispatch(setShowProgressTableV2(true));
 
-    expect(wrapper.find(SectionProgress)).to.have.length(0);
-    expect(wrapper.find(SectionProgressV2)).to.have.length(1);
-    expect(wrapper.find('Link')).to.have.length(1);
-    expect(wrapper.find('Link').at(0).props().children).to.equal(
-      'Switch to old progress view'
-    );
+    screen.getByText(V2_PAGE_LINK_TEXT);
+    screen.getByTestId(V2_TEST_ID);
+
+    expect(screen.queryByText(V1_PAGE_LINK_TEXT)).to.not.exist;
+    expect(screen.queryByText(V1_TEST_ID)).to.not.exist;
   });
 
-  it('shows default if no user preference', () => {
-    const wrapper = setUp({showProgressTableV2: undefined});
+  it('shows default v1 if no user preference', () => {
+    renderDefault();
+    store.dispatch(setShowProgressTableV2(undefined));
 
-    expect(wrapper.find(SectionProgress)).to.have.length(1);
-    expect(wrapper.find(SectionProgressV2)).to.have.length(0);
+    screen.getByText(V1_PAGE_LINK_TEXT);
+    screen.getByTestId(V1_TEST_ID);
+
+    expect(screen.queryByText(V2_PAGE_LINK_TEXT)).to.not.exist;
+    expect(screen.queryByText(V2_TEST_ID)).to.not.exist;
+  });
 
+  it('shows default v2 if no user preference', () => {
     DCDO.set('progress-table-v2-default-v2', true);
-    const wrapper_1 = setUp({showProgressTableV2: undefined});
+    store.dispatch(setShowProgressTableV2(undefined));
+    renderDefault();
 
-    expect(wrapper_1.find(SectionProgress)).to.have.length(0);
-    expect(wrapper_1.find(SectionProgressV2)).to.have.length(1);
+    screen.getByText(V2_PAGE_LINK_TEXT);
```

Comment:
Notes to anyone who is looking at this as an example migration:
* These `queryByText` should probably be `queryByRole('link', {name: 'V#_PAGE_LINK_TEXT'})` to be more accessible and check that they are actually set as links
* All `screen.queryByText(V1_TEST_ID)` should actually be `screen.queryByTestId(V1_TEST_ID)` :) fixed in a follow up PR